Improve "Why?" and "What?" Section of README.md #9 (#22)
dwyl/product-roadmap: README.md

<div align="center">

# Product Roadmap

![roadmap-1152x672-flip](https://user-images.githubusercontent.com/194400/27068309-5b379538-5007-11e7-99fb-20f09369c672.jpg)

</div>
<br />

> # _Why_? 🤷‍♀️

Reviewer: because the hierarchy of the headings (from "##" to "#") has changed, we also need to update the other sections (Who, When, How)

Author: @SimonLab agreed. If you have time, please update. 📝

Reviewer: It should be fixed with https://github.com/dwyl/product-roadmap/pull/22/commits/0862462ef4d7c39f04bead23ecb6a55d08a19c1a

Author: Cool. I will ignore the notifications till you're done. 👍

Reviewer: :+1: I think I'm almost done with formatting. I've been reading the document and I'll add comments/issue focused on the content itself.

Author: Defo ask questions and even open issues for things that are unclear. 👍

Author: Great exploratory question: https://github.com/dwyl/product-roadmap/issues/27 Thanks @SimonLab 🎉

Having a _clear_ **Product Roadmap** is ***essential***
to help ***everyone*** _focus_
on our _collective **goal**_.

> **Note**: this document is a
["work-in-progress"](https://en.wiktionary.org/wiki/work_in_progress#Noun),
it's _definitely_ not "finished". <br />
A _good_ product roadmap evolves over time
in response to _feedback_;
by definition it's _never_ "complete". <br />
What you are reading is a _snapshot_
of the roadmap as it is _right now_.
If you ⭐ (bookmark) the repo 
and return to it in future, it will _definitely_ be different
because we update it fairly regularly.
We use a Git repo so that we (_and anyone `else` in the community_)
can _track_ the changes/evolution as we 
[Learn, Build and Measure](http://theleanstartup.com/principles). <br />
We welcome any/all feedback on it 
_especially_ from people who are `new`
to the 
[`@dwyl`](https://github.com/dwyl/start-here#who) 
community. <br />
Please 
[open an issue](https://github.com/dwyl/product-roadmap/issues)
if you have any ideas/questions.
Thanks! ✨ <br />

<br />

## _Why_ Are We Building An _App_?

If there was an _existing_ 

> (Open Source Cross-platform Real-time Collabroative) 

Reviewer: ```suggestion
(Open Source Cross-platform Real-time Collaborative) 
```

system 
that allowed us to achieve _everything_ we have in mind for the dwyl App,
we would _use_ it without hesitation.
Sadly, we have not found any platform 
that has a
[_systematic_](https://en.wikipedia.org/wiki/System), 
[goal oriented](https://en.wikipedia.org/wiki/Goal_orientation)
and 
[collaborative](https://en.wikipedia.org/wiki/Collaboration)
way to _effectively_ let individuals and/or teams:

1. **Set** **goals** that are ["SMART"](https://en.wikipedia.org/wiki/SMART_criteria)
either freeform or using the [OKR](https://en.wikipedia.org/wiki/OKR) framework.

2. **Capture** everything on one's mind 
as (plaintext) `items`, (voice) `recordings` or `images`
to declutter thoughts 
and think more clearly/creatively

3. **Categorise** the `items` into logical categories using `tags`
(_ideally linked to specific goals_).

4. **Transform** `items` of text into _actionable_ `items` 
such as: 
"Task" (_with context, deadline, progress, dependencies, etc._), 
"Reminder", 
"Calendar" (_Meeting, Date, Event, Travel, etc._), 
"Ingredient" (Shopping list `item`), etc. 

> We want _complete_ versaility in the `tags` that can be appplied to items

Reviewer: ```suggestion
We want _complete_ versatility in the `tags` that can be applied to items
```

so that our App can be used for _any_ purpose that helps people organise their lives.

> 5. ***Track*** tasks and goal **progress _visually_**.

Reviewer: Keep same Emphasis as previous points
```suggestion
5. **Track** tasks and goal progress **_visually_**.
```

Author: Same again. the "Track" word is an _action_ that's why it's italicised. Happy to remove and revisit. 👍

> 6. ***Communicate*** with others what task we are focussing on

Reviewer: ```suggestion
6. **Communicate** with others what task we are focussing on
```

Author: Same here: "Communicate" is an _action_ word. So italics emphasise it. But I _agree_ that I use italics _way_ too often ... 🙄

so we can all avoid interrupting/distracting each other during 
["Deep Work"](https://www.calnewport.com/books/deep-work)
Avoiding the "what you doing" type questions,
or general interruptions is essential for focus work.

> 7. ***Delegate*** specific tasks 

Reviewer: ```suggestion
7. **Delegate** specific tasks 
```

Author: Ah, the reason this italicised is that it's an _action_. We need a style guide. 💭 
Meanwhile applying your suggestion for consistency. 👍

with built-in tracking to _eliminate_
["status update requests"](https://en.wikipedia.org/wiki/Busy_work)

There are _plenty_ of "productivity" apps 
that _partially_ solve these problems.
_Most_ of the Apps you/we can list are Closed Source,
which rules them out (for us) 
for the reasons described in "Why Open Source?" (_below_).
We have _scoured_ GitHub/GitLab/etc. to find an Open Source App/system
we can use and nothing comes _close_ to solving the problem.
We have _extensively studied_ the 
UI/UX 
of the Closed Source Apps 
see: https://github.com/dwyl/product-ux-research/issues
and _none_ of them have the workflow (UX) we want
or they focus on the wrong things!

[![notion-encourages-busywork](https://user-images.githubusercontent.com/194400/85288902-75077480-b48e-11ea-8b41-75299936985b.png)](https://github.com/dwyl/product-ux-research/issues/5#issuecomment-647495254 "Notion encourages busy-work - Click to read")

We want to help people focus on the _one_ `task` 
they need to be doing _right now_,
and easily hide/show the rest when needed.
We want to apply both the science and art of distraction-free focus
we have read about in books such as 
"Deep Work",

> "Indestractable", 

Reviewer 2: ```suggestion
"Indistractable", 
```

Author: Good catch. Thanks for fixing. 👍 
Ref: https://www.amazon.com/Indistractable-Control-Your-Attention-Choose/dp/194883653X

This list is intended as a guide not a "gospel"._
-->

dwyl's
[mission](https://github.com/dwyl/start-here/blob/6fac8f0e4b9a81cba929eb7e577f22a4a0eb734c/mission.md#what-is-dwyls-mission) 
is to:

## _Empower people to maximise effectiveness, creativity and happiness._

> Few people know more about effectiveness 

Reviewer 2: ```suggestion
```

> than life coach and motivational speaker 

Reviewer 2: ```suggestion
```

> [Tony Robbins](https://en.wikipedia.org/wiki/Tony_Robbins). <br />

Reviewer 2: ```suggestion
```

> A quote of his that guides us in our mission is:

Reviewer 2: Tony Robbins has become somewhat controversial because there are dozens of sexual harassment and abuse of power claims against him so I would be tempted to tone this down a bit.

```suggestion
Although Tony Robbins is a somewhat polarising character, we unreservedly agree with this thought:
```

Reviewer 2: Tony Robbins has become somewhat controversial because there are dozens of sexual harassment and abuse of power claims against him so I would be tempted to tone this down a bit as suggested above just so it's not distracting.

Author: Yeah, this makes me very sad. 😢 
+ https://www.buzzfeednews.com/article/katiejmbaker/tony-robbins-accused-sexual-assault-teenager-supercamp
+ https://graziadaily.co.uk/life/opinion/tony-robbins-sexual-assault-and-harassment-allegations/

We can just remove the section or sub it for an alternative... 💭

We believe that by helping people 
_effectively_ keep track 
of their progress 
toward their goals, <br />
they will naturally be happier. 
Happiness is almost a _byproduct_ of personal effectiveness!

## "_Happiness is not something ready made. <br />It comes from your own actions._" ~ [Dalai Lama](https://www.goodreads.com/quotes/1263-happines)

By helping people to focus their efforts
on their highest priority items,
we are helping people be _happy_. <br />
That's a mission we can _all_ get on board with!

Our immediate plan is to build a **_single_ app**
that satisfies _several_ human needs
while focussing on the theme of "***personal effectiveness***".

The @dwyl App will be a "_hybrid_"
containing several features
that together help the individual
achieve _all_ of their personal goals
through a systematic workflow.

<br />

## _What Problem_ Are We _Solving_?

We are building an App that 
creates a systematic approach 
to solve the problem we _all_ face every day. <br />
What do you think of when you 
see/read/hear/think of 
the following words:

## "_I don't have enough time to_..."

How many times have you heard 

> a friend, family member or work colleage 

Reviewer: ```suggestion
a friend, family member or work colleague 
```

because we _feel_ like there isn't enough time. <br />

Whatever your mindset regarding the amount of time 
you have available
to do everything you want in life, 
one thing is certain: **without** a **_systematic_ approach**
we are **_all_ ineffective** in how we spend/invest our **time**.

You may have seen the following quote online:

## "_Lack of direction, not lack of time, is the problem. <br />We all have 24-hour days._" ~ [Zig Ziglar](https://www.goodreads.com/quotes/270489-lack-of-direction-not-lack-of-time-is-the-problem)

or 

### "_The bad news is time flies. The good news is you’re the pilot_” ~ [Michael Altshuler](https://www.goodreads.com/quotes/144299)

or the most 
[cliché](https://en.wikipedia.org/wiki/Clich%C3%A9) of all:

## "_You have the same number of hours in a day as Beyonce_" ~ [Instagram](https://github.com/dwyl/home/issues/29)

The problem with these "motivational" quotes
often repeated on social media 
is that they can make us **feel _worse_**
about our time-management - _**self**_ 
"**productivity** [***shaming***](https://en.wikipedia.org/wiki/Shame)" -
and **don't** actually **_help_** us **move _forward_**!

> In the case of the one mentioning Beyonce, it's an outright _lie_.

Reviewer 2: ```suggestion
```

Reviewer 2: I don't think this is necessary given the following sentence.

**Queen Bey** has an _army_ of people helping her get the most out of her day.

> The rest of us don't have personal assitants and entourages of people

Reviewer: ```suggestion
The rest of us don't have personal assistants and entourages of people
```

but the universal fact is that
very few people are living their _best_ life.
For most people it's not their fault.
The existing _systems_ 
for keeping us "on track" are _fundamentally_ broken.
The pull of distraction is overwhelming to most people.

Billions of people have been lured into _burning_ time 
on low-value entertainment activities 
such as watching TV or 
endlessly scrolling through social media apps
instead of working on their personal goals.

> As former Facebook President, 
[Sean Parker](https://github.com/dwyl/learn-react/issues/23#issuecomment-647521076) 
put it: 
Facebook's goal is to <br />
"_consume as much of your time and conscious attention as possible_". <br />
> The more time you spend on their platforms
the more money they all make. <br />
They don't care that they are _systematically_
[wasting the lives of millions of people](https://github.com/nelsonic/nelsonic.github.io/issues/779#issuecomment-625188176).

<!--
> For the single mom with 2 kids and 3 jobs to make ends meet
who doesn't spend a single _second_ on entertainment,
our App will _still_ be able to help!
TODO: Create Persona for this person: https://github.com/dwyl/app/issues/213
-->

> The _reason_ most people are _innefective_ at focussing

Reviewer: ```suggestion
The _reason_ most people are _ineffective_ at focussing
```

on tasks that move them toward a specific goal
is that they don't have adequate _systems_
for focussing on what matters 
and maximising personal effectiveness.
Meanwhile the _system_ 
[Mark & Pals](https://github.com/dwyl/learn-react/issues/23) 
have created has an army of 
Social/Computer Science PhDs 
whose job it is to make their time-wasting Apps
as [psychologically addictive](https://www.theguardian.com/technology/2018/may/08/social-media-copies-gambling-methods-to-create-psychological-cravings) 
as possible.
The average person without a systematic approach doesn't stand a _chance_!

### System

## "_You do not rise to the level of your goals. <br /> You fall to the level of your systems._" <br />~ [James Clear](https://www.goodreads.com/quotes/9536717), Atomic Habits

The ***system*** we are building

> is desgined to help ourselves and others 

Reviewer: ```suggestion
is designed to help ourselves and others 
```

move towards our
(life/health/fitness/career/etc.) 

> goals is what we are building.

Reviewer 2: ```suggestion
goals.
```

Reviewer 2: Repeated from the beginning of the sentence

We haven't found a catchy one-line phrase

> to summarise the ***system*** we are building. ([`#HelpWanted`]())

Reviewer 2: ```suggestion
to summarise it yet. ([`#HelpWanted`]())
```

What do _you_ need to (_politely and plainly_) say "***no***" to 
so you can **focus** on your **goals**?

<!-- 
https://zenhabits.net/enough-time
https://www.lifehack.org/articles/productivity/10-reasons-why-some-people-feel-like-they-dont-have-enough-time.html
-->

<br />

## 3 Stages of Our Roadmap

### 1. _Individual_ Person Experience -> [_Single Player_](https://en.wikipedia.org/wiki/Single-player_video_game)

If the App is not _immediately_ useful to an _individual_ person 
and relies on "network effects" it will be 
[DOA](https://en.wikipedia.org/wiki/Dead_on_arrival).  <br />
We need to _optimise_ for the individual _first_ 
and then once we are able to use the App individually _ourselves_, 
we can move on to the "Team" experience (see below). 

In the _individual_ person experience, the person uses the dwyl App to:

+ **capture** > **brain-dump** everything they have in their head 
so that they can think clearly (_without a cluttered mind_) and prioritise. 📝 
  
  > Capturing text (_and later voice memos and images_) as "items" 
  which can then be categorised/organised using tags 

>   is the most intiuitive and versatile way of storing data.

Reviewer: ```suggestion
  is the most intuitive and versatile way of storing data.
```

  We only need _one_ field or data type (`text`).
  We can then apply `tags` to an `item` to make it into _anything_ else.
  And then the UI renders the item with the appropriate UX 

>   for an `item` with the `tag` `"task"`, display a checkbox next to it.

Reviewer: ```suggestion
  for an `item` with the `tag` `"task"`, displaying a checkbox next to it.
```

Author: This is technically acceptable grammar; the action is to "_display_" the checkbox.

+ define **individual goals** so they can be more systematic about achieving them. 
e.g:
  + **Health**: **exercise** for (at least) **1 minute** each day 
  so I can get/stay fit ❤️  
  + **Money**: **save** at least **1%** of my **income** each week/month 
  for a rainy day/retirement 💰
  + **Family**: spend at least 1 minute of quality time 
  with a family member each day  💑  
  + **Sleep**: get a _full_ night's restful sleep 
  each night as the foundation for all other goals. 🛌 
  + **Career**: practice/learn core work **skill(s)** for **1 minute** 
  each day to achieve mastery. 👩‍💻 
  + etc ... 
  
  > There are _many_ areas where people can have life goals. 
  Eventually our app will encourage people 
  to have goals defined for each area of a person's life. 
  We will have an on-boarding questionnaire to guide them, 
  a visualisation of "progress" toward each one of the goals 

>   and reminder to review them regularly. 

Reviewer: ```suggestion
  and reminders to review them regularly. 
```

  If someone wants to use our app just 
  to take random  `plaintext` notes that's 100% OK. 
  We will write Blog posts (_product release notes_) 
  for the features we build in the future 
  that will _guide_ people toward having 
  a more systematic approach. 
  e.g: set goals, write everything down, 
  tie all tasks to specific goals, etc. 
  But until we _build_ those features into our App, 
  we _must_ make it _useful_ in `plaintext` mode.

+ ***prioritise*** the `items` that need to be done first,
those that can be differed or delegated.

  > "_If You Don’t Prioritize Your Life, Someone Else Will_". 
  ~ [Greg McKeown](https://hbr.org/2012/06/how-to-say-no-to-a-controlling), 
  [Essentialism](https://youtu.be/euJc0UmSik8)

+ ***Use*** the App to stay "on track" 
toward _personal_ goals 
and get the most important things done.

<br />

While we feel the _individual_ experience of the App
needs to be flawless as we want anyone 
to be able to use the app in isolation,
we feel the App becomes _exponentially_ more useful/powerful in _teams_.

<br />

### 2. Small _Team_ Experience -> [_Co-op Mode_](https://en.wikipedia.org/wiki/Cooperative_gameplay)
 

> We feel that our App will be _incredibly_ 

Reviewer: ```suggestion
We feel that our App will be _incredibly_ helpful
```

to small "teams" of people who need to collaborate on getting things done.
Small teams will not be limited to "work" colleagues.
Rather we will have personas for families, groups of friends and other associations. 
We will focus on this "team" use-case 
as soon as we have the basic _individual_ UX working.

> If you are keen on helping us with the Colab features,
please take a look at our existing backlog: 
https://github.com/dwyl/app/issues 
If there is a feature you don't see and want to help us describe, 
please open an issue!

> ## 3. Big Company/Org Experience -> [_MMORPG_](https://en.wikipedia.org/wiki/Massively_multiplayer_online_role-playing_game)!

Reviewer: Third point of the stages of the roadmap should have the same type of title to the two previous points
```suggestion
### 3. Big Company/Org Experience -> [_MMORPG_](https://en.wikipedia.org/wiki/Massively_multiplayer_online_role-playing_game)!
```

There is no reason why a "Big" company or organisation cannot use the dwyl App.
But this is not our focus right now. 

> We need to crawl, walk & run before we attempt to _fly_. 

Reviewer 2: 👍

Author: 👍

achieve _all_ of their personal goals
through a systematic workflow.

> ## 1. _Individual_ Person Experience -> [_Single Player_](https://en.wikipedia.org/wiki/Single-player_video_game)

Author: I'm busy expanding on the **_Individual_ Person's Experience** as the _first_ thing we are building.
related to: https://github.com/dwyl/app/issues/213 (Create Personas)

Author: We can add personas in the next PR.

For the first step (***Capture***),
we will build an ultra-simple interface
to allow people to

> input **`palintext`**

Reviewer: ```suggestion
input **`plaintext`**
```

and save it to a backend as **`text`**

> in one ore more **`items`**.

Reviewer: ```suggestion
in one or more **`items`**.
```

we will build a simple checklist system.
See: [`checklists.md`](https://github.com/dwyl/product-roadmap/blob/master/checklists.md)

<br />

### USP: What is _different_ about the @dwyl App?

A valid/relevant **question** is: 
what makes the @dwyl App unique?

i.e. What is our
["unique selling point"](https://en.wikipedia.org/wiki/Unique_selling_proposition)?<br />
The **answer** is: 
_initially_ it may _appear_ that our App

> does not have much different

Reviewer: ```suggestion
does not differ much
```

+ [ ] Get "Sign-off" (_Confirmation_) / Feedback on Completed Task
-->

<hr />

> **Note**: while the App Store might categorise 
the App we are building as 
["Productivity"](https://apps.apple.com/us/genre/ios-productivity/id6007),
we don't refer to it as 

## Why Open Source?

## _Public_?

> From a "commercial" perspective making our

Reviewer: ```suggestion
From a "commercial" perspective, making our
```

_Committing to a long-range map implies a fixed/inflexible destination._

+ _**Doubt** in their **own ability** to **deliver** the plan_.

<br /><hr />

<!--
[![HitCount](http://hits.dwyl.com/dwyl/product-roadmap.svg)](http://hits.dwyl.com/dwyl/product-roadmap) -->

### Final Note

The purpose of this document is to 
share the "Vision" for what we are building.
No map is a _complete_ representation of the world;
it's just a "model" that allows us to think about it.
When you open Google Maps on your phone

> it does not show you all the street lights and advertising bilboards.

Reviewer: ```suggestion
it does not show you all the street lights and advertising billboards.
```

It's the same thing with a Software Product Roadmap.
Including _everything_ would be totally overwhelming to the reader.

Just because a feature is _not_ on this roadmap
does not mean we have not thought of it.
In some cases ideas are only in the Backlog: 
[app/issues](https://github.com/dwyl/app/issues) 
and are not reflected here.
In _many_ cases the features/ideas
are added to and discussed on the backlog
and never added to Roadmap because 
they don't _need_ to be on the map.

Our long-term objective is to build a "Platform" for Personal Effectiveness
that _other_ people can add their own features to. 
That is not described here 
because it's too "grandiose" at this stage.
Instead, that part of our long-term vision
is summarised as "API". 
Having a REST+Realtime API will allow

> anone to inteact with their data 

Reviewer: ```suggestion
anyone to interact with their data 
```
